app/models/service_template_transformation_plan_task.rb

```diff
@@ -114,6 +170,22 @@ def canceled
     update_attributes(:cancelation_status => MiqRequestTask::CANCEL_STATUS_FINISHED)
   end
 
+  def conversion_options
+    source_cluster = source.ems_cluster
+    source_storage = source.hardware.disks.select { |d| d.device_type == 'disk' }.first.storage
+    destination_cluster = transformation_destination(source_cluster)
+    destination_storage = transformation_destination(source_storage)
+
+    [
+      {
+        :source_disks     => source_disks.map { |disk| disk[:path] },
+        :network_mappings => network_mappings
+      },
+      send("conversion_options_source_provider_#{source_ems.emstype}_#{source_transport_method}", source_storage),
+      send("conversion_options_destination_provider_#{destination_ems.emstype}", destination_cluster, destination_storage)
+    ].inject(&:merge)
```

Comment:
The array and `inject(&:merge)` is more complicated than it needs to be

Can you keep it simple?
```
conversion_options_source_provider_meth = "conversion_options_source_provider_#{source_ems.emstype}_#{source_transport_method}"
conversion_options_dest_provider_meth   = "conversion_options_destination_provider_#{destination_ems.emstype}"

options = {
  :source _disks    => source_disks.map { |disk| disk[:path] },
  :network_mappings => network_mappings
}

options.merge(send(conversion_options_source_provider_meth, source_storage))
options.merge(send(conversion_options_dest_provider_meth, destination_cluster, destination_storage))

options
```

It might not be as slick but it takes less brainpower to figure out what is going on if you're just scanning.